libs/SalesforceSDK/src/com/salesforce/androidsdk/ui/OAuthWebviewHelper.kt

```diff
@@ -660,9 +667,16 @@ open class OAuthWebviewHelper : KeyChainAliasCallback {
                         null
                     )
 
-                    else -> when {
-                        instance.useWebServerAuthentication -> onWebServerFlowComplete(params["code"])
-                        else -> onAuthFlowComplete(TokenEndpointResponse(params))
+                    else -> {
+                        // Determine if presence of override parameters require the user agent flow.
+                        val overrideWithUserAgentFlow = isUsingFrontDoorBridgeUrl && frontDoorBridgeCodeVerifier == null
```

Comment:
When would this occur?

Reply:
This occurs when the QR code provides a front door bridge URL but _not_ a code verifier, which implicitly sends us to user agent flow.  That's my understanding after chatting with @wmathurin.  Possibly one of most interesting points in all of this is ensuring this dovetails well with out existing preferences between user agent and web server flow.

I've tried quite a few combinations to test it out, such as changing the app's `useWebServerAuthentication` setting plus using QR codes of both preferences.  It matches my understanding of it thus far, but I love more input if there's any to have.

Reply:
I guess we could instead fail if the configured auth flow does not match the auth flow the QR URL implies. And show a message to the user e.g. "QR URL is for a user-agent flow but mobile application is configured for web server flow".

Reply:
Actually, this switch to the flow in the QR code is working for both Android and iOS.  It's a one-time override, since a success or failure will reset everything back to the flow set by the app.